simulator.py

```diff
@@ -0,0 +1,41 @@
+import numpy
+
+
+class Simulator(object):
+    def __init__(self, state_space, action_space, discount):
+        self.state_space = state_space
+        self.action_space = action_space
+        self.discount = discount
+        self.max_iterations = 1000
+        self.tolerance = 0.01
```

Comment:
These seem like they should be constructor parameters, maybe with default values if you don't feel like setting them elsewhere.